app/models/search_document.rb

```diff
@@ -0,0 +1,18 @@
+class SearchDocument < ActiveRecord::Base
+  belongs_to :searchable, polymorphic: true
+
+  # cant quite believe that sql is working...
+  def self.search(query, team_id)
+    where(team_id: team_id).joins('
+      INNER JOIN (SELECT
+        "search_documents"."id" AS search_id,
+        (ts_rank((to_tsvector(\'english\', coalesce("search_documents"."content"::text, \'\'))),
+        (to_tsquery(\'english\', \'\'\' \' || \'' + query + '\' || \' \'\'\' || \':*\')), 0)) AS rank
```

Comment:
`query` is gonna need some escaping to ensure there isn’t any SQL injection going on. Alongside stopping that I believe there are also special characters when using PG search like `*` etc may need to check that and strip them out as appropriate